File: app/src/main/java/com/prateek/boardcastactivitydemo/MainActivity.java
package com.prateek.boardcastactivitydemo;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void broadcastIntent(View view){
        Intent intent = new Intent();
        intent.setAction("com.prateek.CUSTOM_INTENT");
        sendBroadcast(intent);
    }
}

File: app/src/main/java/com/prateek/boardcastactivitydemo/MainActivity.kt
package com.prateek.boardcastactivitydemo

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun broadcastIntent(view: View) {
        val intent = Intent()
        intent.action = "com.prateek.CUSTOM_INTENT"
        sendBroadcast(intent)

    }
}
